spec/controllers/dashboard_controller_spec.rb

```diff
@@ -2,11 +2,14 @@
 require "rails_helper"
 
 RSpec.describe DashboardController, type: :controller do
-  let(:user)   { create(:user) }
-  let(:minion) { create(:minion) }
+  let(:user)          { create(:user) }
+  let(:minion1)       { create(:minion) }
+  let(:minion2)       { create(:minion) }
+  let(:master_minion) { create(:master_minion) }
+  let(:worker_minion) { create(:worker_minion) }
 
   before do
-    Minion.create! [{ hostname: "master" }, { hostname: "minion0" }]
+    minion1 && minion2
```

Comment:
How would be more readable? The idea is that we `let()` as before, so we don't forcefully create it (some test cases test that there are nothing yet created), and we force them to be created with the factory depending on the context of the test `minion1 && minion2` or `master_minion && worker_minion` (the idea is to remove complexity in the test using `Minion.assign_roles!`, because all we care about is that minions have roles assigned).